### ISSUE TITLE
A pre-existing mapping object with a WordPress ID of 0 will cause ongoing failures.

### ISSUE THREAD

**Issue by jonathanstegall** [MinnPost/object-sync-for-salesforce#78]
In Salesforce_Pull, it works like this:

Starting at line 769.

```
// find out if there is a mapping object for this wordpress object already
$mapping_object = $this->mappings->get_object_maps(
	array(
		'wordpress_id' => $wordpress_id,
		'wordpress_object' => $salesforce_mapping['wordpress_object'],
	)
);

// there is already a mapping object. don't change the wordpress data to match this new salesforce record, but log it
if ( isset( $mapping_object['id'] ) ) {
```

Then we create a log entry. We don't create a new object map, though. So data could be syncing, but the plugin would never realize it.

This only happens on Salesforce Pull I think. But we need to verify this.

**Comment by jonathanstegall** [comment]
Some context:

The plugin creates a mapping object with a WordPress ID of 0 if the actual WordPress object has not yet been created. This is the only time this happens, as far as I know.

After the result comes back from the Wordpress class, it gets the ID from the data result and updates the mapping object row.

There have been some previous fixes to get to this point. It's possible that this bug was caused by older database rows, but it's also possible that there are still rows with WordPress ID of 0 that are being created and never updated.

The above fix will keep their existence from stopping new additions. But that is only a partial fix, I think. We need to find out in which cases the 0 never gets updated.

**Comment by jonathanstegall** [comment]
This means we're maybe still kind of dealing with #59.

**Comment by jonathanstegall** [comment]
This is also addressed in 4b63e0beccbe987fa13f4aa722826efcbf7b3bac
